crates/api/src/gossip.rs

```diff
@@ -18,13 +22,17 @@ pub trait GossipFactory: 'static + Send + Sync + std::fmt::Debug {
     fn default_config(&self, config: &mut config::Config) -> K2Result<()>;
 
     /// Construct a gossip instance.
+    #[allow(clippy::too_many_arguments)]
     fn create(
         &self,
         builder: Arc<builder::Builder>,
-        space: SpaceId,
+        space_id: SpaceId,
+        space: DynSpace,
```

Comment:
I'm not sure this can work. The space is going to be constructing the gossip instance, and while we could potentially use [Arc::new_cyclic](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.new_cyclic) to keep a weak reference to ourselves to pass into the gossip module, I'd worry about that type of circularity causing mutex deadlocks.

Reply:
Relying to this comment and the one below. Yes I had some trouble with circular references and that's why I'm holding a weak reference to it. I think splitting the local agents into their own little store would be a solution but I'm wondering how likely deadlocks are because of this? The only thing I was really concerned about was de-allocation working properly when both sides were using full arcs

Reply:
Okay, cool. I think a cloned local agent data type would be safer and easier to maintain, but I'm also fine with this and testing, especially as far as merging this pr goes. We can always update as a followon if needed.